From b67c501336bb60e18d554007b8a4150fe1e88739 Mon Sep 17 00:00:00 2001
From: Babu Shanmugam <bschanmu@redhat.com>
Date: Tue, 15 Nov 2016 11:42:12 +0000
Subject: [PATCH 3/3] OVN plugin configuration fixes

This patch includes setting additional configurations for the neutron
plugins for ovn.

Change-Id: Idc9e7ef4a1b88013bca3eac3c136e4710e38a5c0
---
 puppet/services/neutron-compute-plugin-ovn.yaml |  1 -
 puppet/services/neutron-plugin-ml2-ovn.yaml     | 17 +++++++++++++----
 2 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/puppet/services/neutron-compute-plugin-ovn.yaml b/puppet/services/neutron-compute-plugin-ovn.yaml
index 95e05dd..fa0a78b 100644
--- a/puppet/services/neutron-compute-plugin-ovn.yaml
+++ b/puppet/services/neutron-compute-plugin-ovn.yaml
@@ -37,7 +37,6 @@ outputs:
     value:
       service_name: neutron_compute_plugin_ovn
       config_settings:
-        tripleo::profile::base::neutron::agents::ovn::ovn_db_host: {get_param: OVNDbHost}
         ovn::southbound::port: {get_param: OVNSouthboundServerPort}
         ovn::southbound::encap_type: {get_param: OVNTunnelEncapType}
         ovn::controller::ovn_encap_ip: {get_param: [ServiceNetMap, NeutronApiNetwork]}
diff --git a/puppet/services/neutron-plugin-ml2-ovn.yaml b/puppet/services/neutron-plugin-ml2-ovn.yaml
index 20dfda6..ef00a9c 100644
--- a/puppet/services/neutron-plugin-ml2-ovn.yaml
+++ b/puppet/services/neutron-plugin-ml2-ovn.yaml
@@ -18,6 +18,10 @@ parameters:
     description: Mapping of service endpoint -> protocol. Typically set
                  via parameter_defaults in the resource registry.
     type: json
+  OVNSouthboundServerPort:
+    description: Port of the OVN Southbound DB server
+    type: number
+    default: 6642
   OVNDbConnectionTimeout:
     description: Timeout in seconds for the OVSDB connection transaction
     type: number
@@ -61,10 +65,15 @@ outputs:
       config_settings:
         map_merge:
           - get_attr: [NeutronMl2Base, role_data, config_settings]
-          - neutron::plugins::ovn::ovsdb_connection_timeout: {get_param: OVNDbConnectionTimeout}
-            neutron::plugins::ovn::neutron_sync_mode: {get_param: OVNNeutronSyncMode}
-            neutron::plugins::ovn::ovn_l3_mode: true
-            neutron::plugins::ovn::vif_type: {get_param: OVNVifType}
+          - ovn::southbound::port: {get_param: OVNSouthboundServerPort}
+            neutron::plugins::ml2::ovn::ovsdb_connection_timeout: {get_param: OVNDbConnectionTimeout}
+            neutron::plugins::ml2::ovn::neutron_sync_mode: {get_param: OVNNeutronSyncMode}
+            neutron::plugins::ml2::ovn::ovn_l3_mode: true
+            neutron::plugins::ml2::ovn::vif_type: {get_param: OVNVifType}
             neutron::server::qos_notification_drivers: {get_param: OVNQosDriver}
+            neutron::plugins::ml2::type_drivers: [ 'geneve', 'vlan', 'flat' ]
+            neutron::plugins::ml2::tenant_network_types: [ 'geneve' ]
+            neutron::plugins::ml2::max_header_size: 38
+            neutron::service_plugins: ['qos', 'networking_ovn.l3.l3_ovn.OVNL3RouterPlugin']
       step_config: |
         include ::tripleo::profile::base::neutron::plugins::ml2
-- 
2.7.4

